crates/observe/src/tracing_reload_handler.rs

```diff
@@ -21,6 +21,7 @@ pub(crate) fn spawn_reload_handler<T: 'static>(
 
         let socket_path = format!("/tmp/log_filter_override_{name}_{id}.sock");
         tracing::warn!(file = socket_path, "open log filter reload socket");
+        std::fs::remove_file(&socket_path).ok();
```

Comment:
nit: Let's not block the async run time unnecessarily with blocking functions.
Also if we want to ignore a return value I think it's clearer to use `let _ = result` instead of `result.ok()`.
```suggestion
        let _ = tokio::fs::remove_file(&socket_path).await;
```